pkg/azurefile/controllerserver.go

```diff
@@ -895,12 +902,12 @@ func (d *Driver) DeleteSnapshot(ctx context.Context, req *csi.DeleteSnapshotRequ
 		return nil, status.Error(codes.InvalidArgument, "Snapshot ID must be provided")
 	}
 
-	shareURL, err := d.getShareURL(ctx, req.SnapshotId, req.GetSecrets())
```

Comment:
`if len(req.GetSecrets()) > 0` we still need to use data plane api, and if `if len(req.GetSecrets()) == 0`, then use management api.